curvefs/src/metaserver/inode_storage.cpp

```diff
@@ -156,35 +165,167 @@ MetaStatusCode InodeStorage::GetXAttr(const Key4Inode& key, XAttr *xattr) {
 }
 
 MetaStatusCode InodeStorage::Delete(const Key4Inode& key) {
-    WriteLockGuard writeLockGuard(rwLock_);
+    WriteLockGuard lg(rwLock_);
     std::string skey = conv_.SerializeToString(key);
-    if (!FindKey(skey)) {
-        return MetaStatusCode::NOT_FOUND;
-    }
-
-    Status s = kvStorage_->HDel(table4inode_, skey);
+    Status s = kvStorage_->HDel(table4Inode_, skey);
     if (s.ok()) {
-        EraseKey(skey);
+        // NOTE: for rocksdb storage, it will never check whether
+        // the key exist in delete(), so if the client delete the
+        // unexist inode in some anbormal cases, it will cause the
+        // nInode less then the real value.
+        nInode_--;
```

Comment:
You need check whether others need the exact value.

Reply:
> You need check whether others need the exact value.

yep, i had check it.